src/classes/REL_Relationships_TEST.cls

```diff
@@ -226,74 +217,105 @@ public class REL_Relationships_TEST {
+    */
+    @isTest
+    public static void testCampaignMemberReentrancyCheck(){
+        setupRelationshipTestData(null);
+
+        // Install Default HEDA TDTM Configuration
+        List<TDTM_Global_API.TdtmToken> defaultTokens = TDTM_Global_API.getDefaultTdtmConfig();
+        TDTM_Global_API.setTdtmConfig(defaultTokens, 'hed');
+
+        // Add the Test TDTM for the Relationship object. This Test TDTM will simply update the campaign member that caused the relationship to be inserted.
+        // So the flow would be: campaign member updated -> relationship updated -> campaign member updated again
+        Trigger_Handler__c newTDTM = new Trigger_Handler__c(Active__c = true, Asynchronous__c = false,
+            Class__c = 'UTIL_UnitTestData_TEST.Test_TDTM', Load_Order__c = 1, Object__c = 'Relationship__c',
+            Owned_by_Namespace__c = 'hed', Trigger_Action__c = 'AfterInsert;AfterUpdate');
+        insert newTDTM;
+
+        //Insert a campaign
+        Campaign campaign = new Campaign(Name='TestCampaign', IsActive=true, Type='Other');
+        insert campaign;
+
+        //Insert a campaign member
+        CampaignMember cm = new CampaignMember(ContactId = c2.Id, CampaignId = campaign.Id);
+        insert cm;
+
+        // Ensure that the campaign member gets updated recursively
+        UTIL_UnitTestData_TEST.sObjectsToUpdate = new List<SObject>{ cm };
+
+        Test.startTest();
+        // Turn the reentrancy flag on since the class won't actually insert any relationships, so the flag would remain off otherwise.
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.REL_Relationships_Cm_TDTM, true);
+        cm.Status = 'SomethingDifferentForUpdate';
+        update cm;
+        Test.stopTest();
+
+        System.assert(REL_Relationships_Cm_TDTM.reentrancyPrevented == true, 
+            'Expected the reentrancy check to have prevented the REL_Relationships_Cm_TDTM trigger from processing again.');
+    }
+
     /*********************************************************************************************************
     * @description Test method for campaign member auto create relationships on update.
     * We can't do a true test as ContactId is not updatable, but we'll hit the update for coverage.
     */
     @isTest
     public static void testAutoCreateCampaignMemberUpdate(){
         setupRelationshipTestData(null);
-        //reset the process controller to mimic a second user action
-        //Relationships_ProcessControl.hasRun = false;
         //insert a campaignmember
-        Campaign c = new Campaign(Name='TestCampaign', isActive=true);
+        Campaign c = new Campaign(Name='TestCampaign', IsActive=true);
         insert c;
-        CampaignMember cm = new CampaignMember(ContactId = c2.id, CampaignId = c.id);
+        CampaignMember cm = new CampaignMember(ContactId = c2.Id, CampaignId = c.Id);
```

Comment:
more of the argument spacing things. I'm not recommending one or another (I prefer no space) but we should be consistent, especially since these lines are right next to each other and two different styles.